File: news_server/modules/news/presentation/http/registry.ts
/**
 * HTTP Handler 注册表
 * 类似 Sjgz-Backend 的 interfaces/http/registry.go
 */
import { NewsHandler } from './handler/news'

export class Registry {
  public readonly News: NewsHandler

  constructor(newsHandler: NewsHandler) {
    this.News = newsHandler
  }
}

File: modules/news/interfaces/http/registry.go
package http

import (
	authconn "nfxnews/connections/auth"
	newsapp "nfxnews/modules/news/application/news"
	"nfxnews/modules/news/interfaces/http/handler"
)

type Registry struct {
	App  *handler.NewsHandler
	I18n *handler.I18nHandler
}

func NewRegistry(svc *newsapp.Service, langs string, identity *authconn.Client) *Registry {
	return &Registry{App: handler.NewNewsHandler(svc, identity), I18n: handler.NewI18nHandler(langs)}
}
